packages/flutter/test/widgets/scrollable_semantics_test.dart

```diff
@@ -255,17 +315,17 @@ void main() {
   });
 }
 
-Future<Null> flingUp(WidgetTester tester, { int repetitions: 1 }) async {
-  while (repetitions-- > 0) {
-    await tester.fling(find.byType(ListView), const Offset(0.0, -200.0), 1000.0);
-    await tester.pump();
-    await tester.pump(const Duration(seconds: 5));
-  }
-}
+Future<Null> flingUp(WidgetTester tester, { int repetitions: 1 }) => fling(tester, const Offset(0.0, -200.0), repetitions);
+
+Future<Null> flingDown(WidgetTester tester, { int repetitions: 1 }) => fling(tester, const Offset(0.0, 200.0), repetitions);
+
+Future<Null> flingRight(WidgetTester tester, { int repetitions: 1 }) => fling(tester, const Offset(200.0, 0.0), repetitions);
+
+Future<Null> flingLeft(WidgetTester tester, { int repetitions: 1 }) => fling(tester, const Offset(-200.0, 0.0), repetitions);
```

Comment:
Not sure if these tests are sensitive to text direction, but should we instead use "forwards"/"backwards" so these work in LTR and RTL? You'd have to pick different direction based on `Directionality`.

Reply:
I think in a test it's probably best to have the input be absolute. You want to test it in both LTR and RTL, but you want to get different results, you don't want the test itself to be agnostic.

Reply:
I find it easier to think about these methods in terms of the direction the finger moves. flinkLeft just means the finger moves towards the left.